File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.java
class Solution {
  public int maxProfit(int[] prices) {
    int ans = 0;
    int l = 0;
    int r = 1;
    while (r < prices.length) {
      if (prices[r] >= prices[l]) {
        while (r < prices.length - 1 && prices[r + 1] >= prices[r]) r++;
        ans += prices[r]-prices[l];
      }
      l = r;
      r++;
    }
    return ans;
  }
}

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.go
func maxProfit(prices []int) int {
  ans, l, r := 0, 0, 1
  for r < len(prices) {
    if prices[r] >= prices[l] {
      for r < len(prices) - 1 && prices[r + 1] >= prices[r] {
        r++
      }
      ans += prices[r] - prices[l]
    }
    l = r
    r++
  }
  return ans
}